onnxruntime/core/providers/cpu/tensor/gather.cc

```diff
@@ -41,7 +41,7 @@ Status GatherCopyData(const Tensor* indices_tensor, const uint8_t* src_base, uin
   // We can't merge this code in the omp loop below as omp does not allow return in the loop
   for (int64_t i = 0; i < N; ++i) {
     Tin idx = indices_data[i];
-    if (idx < 0 || idx >= input_data_shape[axis]) {
+    if (idx < -input_data_shape[axis] || idx >= input_data_shape[axis]) {
```

Comment:
Can you use HandleNegativeAxis in providers/common.h?

Reply:
Good point! Done.

Reply:
Does this exception from HandleNegativeAxis (with its exception text confusingly referring to axis not indices) work properly across what is currently the OMP boundary?

Reply:
@tracysh are you suggesting to change the exception text?